plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -257,4 +257,25 @@ protected static List<Column> toHiveColumns(List<NestedField> columns)
                         Optional.empty()))
                 .collect(toImmutableList());
     }
+
+    protected boolean isMetadataPresent(String metadataLocation)
+    {
+        AtomicReference<Boolean> isSuccess = new AtomicReference<>(false);
+
+        Tasks.foreach(metadataLocation)
+                .retry(3)
+                .suppressFailureWhenFinished()
```

Comment:
Why do we suppress the failure when finished?

I am assuming that you are using https://github.com/apache/iceberg/blob/f02817f6fa1df304b2a1ad2df266406b79a00f0b/core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java#L325-L356 as inspiration.

In case we are dealing with connection errors here they will be suppressed and we'll consider that the metadata file is not present, even though it could actually be present - but we don't know it.

Please rethink this part - probably we do need `3` values as initially implemented - sorry for misleading you.

Reply:
> In case we are dealing with connection errors here they will be suppressed and we'll consider that the metadata file is not present, even though it could actually be present - but we don't know it.

Yes, In this case, we are not going to delete the metadata file and throw `CommitStateUnknownException` in the caller.

Reply:
>  probably we do need `3` values as initially implemented 

@findinpath, Can you please point me to which `3 values`? I did not get it.

Reply:
The outcome of the metadata existence check is either:
- exists
- does not exist
- unknown: when the check operation is experiencing failures (e.g. : connection failure, timeouts) and we can't know with certainty whether the metadata file is referenced or not in the table's metadata.

Reply:
Earlier implementation of check operation returns `SUCCESS` or `UNKNOWN`. It does not tell about failure.

Reply:
Updated code with introducing `CommitStatus` enum.